src/EditorFeatures/CSharpTest/Diagnostics/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -3969,5 +3969,93 @@ private static void TakesParams(params object[] foo)
     }
 }");
         }
+
+        [WorkItem(18510, "https://github.com/dotnet/roslyn/issues/18510")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
+        public async Task DontRemoveCastOnNegatingEnumValue1()
```

Comment:
:bulb: You can use `[Theory]` instead of `[Fact]` here to eliminate some redundancy:

Start by adding a parameter for the operator:

```csharp
public async Task DontRemoveCastOnUnaryOperatorEnumValue(string op)
```

Then replace `Fact` with `Theory` and provide the values for the parameter with `InlineData`:

```csharp
[Theory]
[InlineData("-")]
[InlineData("+")]
[InlineData("~")]
```

Then use `op` with an interpolated string:

```csharp
            await TestMissingInRegularAndScriptAsync(
$@"
enum Sign
    {{
        Positive = 1,
        Negative = -1
    }}

    class T
    {{
        void Foo()
        {{
            Sign mySign = Sign.Positive;
            Sign invertedSign = (Sign) ( [|{op}(int) mySign|] );
        }}
    }}");
```

:memo: If you do this, you would need two tests to cover all 6 cases (3 operators, with and without parentheses).